Enhance Memory Address Formatting in Memory view

planger opened the pull request:
# DO NOT MERGE THIS IS JUST FOR INTERNAL REVIEW

#### What it does

* Add setting for visibility of base prefix (radix) in memory address
* Add setting for memory address format (binary, hex, ...)
* Initialize based on settings for new views
* Add UI to allow changing both options for each view instance
* Style radix prefix with slight opacity for easier reading
* Prepare for adding `0`s in the beginning for 32b or 64b systems but switch padding off by default as we don't know whether debuggee is 32b or 64b

Fixes https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/issues/62

#### How to test
* [ ] Open VS Code settings for Memory inspector
* [ ] Start a debug session
* [ ] Open two Memory views
* [ ] Check whether both have the defaults set for memory address formatting (base prefix and base format)
* [ ] Change both settings in one of the views and make sure it updates correctly only in this single view
* [ ] Click reset button and check whether memory address format settings are reverted
* [ ] Change VS Code settings for base prefix and base format
* [ ] Click reset button again in Memory view to check they update accordingly

#### Review checklist

- [X] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

haydar-metin commented on src/common/memory-range.ts:
```diff
@@ -65,8 +65,33 @@ export function determineRelationship(candidate: bigint, range?: BigIntMemoryRan
+
+const radixPrefixMap: { [key: number]: string } = {
+    [Radix.Binary]: '0b',
+    [Radix.Octal]: 'O',
```
Is'nt the prefix of octal `0o`?

planger replied:
I guess both is common, but for consistency, you are right `0o` is more in line with the others.

haydar-metin reviewed (commented):
Works as intended.

planger committed:
Enhance Memory Address Formatting in Memory view

* Add setting for visibility of base prefix (radix) in memory address
* Add setting for memory address format (binary, hex, ...)
* Initialize based on settings for new views
* Add UI to allow changing both options for each view instance
* Style radix prefix with slight opacity for easier reading
* Prepare for adding 0s in the beginning for 32b or 64b systems
  but switch padding off by default as we don't know whether debuggee
  is 32b or 64b

Fixes https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/issues/62

planger commented:
https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/pull/79
